lib/pacemaker/pcmk_sched_primitive.c

```diff
@@ -544,6 +544,32 @@ create_migration_actions(pe_resource_t *rsc, const pe_node_t *current)
     }
 }
 
+/*!
+ * \internal
+ * \brief Abort a dangling migration by scheduling a stop (and possibly cleanup)
+ *
+ * \param[in]     data       Source node of dangling migration
```

Comment:
If only. A lot of them call functions that expect non-const parameters for a variety of reasons. For example, a lot of the libxml2 functions take non-const arguments, and there's no getting around that other than making a non-const copy. Some kind of "get XML document root" function (I don't remember the name of off the top of my head but I think we define it in pacemaker) creates a document node if one doesn't already exist, so anything that calls that has to be non-const. Some functions don't modify an argument but return a non-const pointer to some piece of it.

Reply:
Yep. Another example is uber_parent(rsc), which will return rsc if rsc does not have a parent. Since most callers need to be able to modify the return value, that means rsc can't be const, even though the function itself doesn't change it. We could work around such cases with a parallel set of functions that take and return const, but that gets complicated.

Reply:
Not worth it